pallets/file-system/src/utils.rs

```diff
@@ -176,10 +206,91 @@ where
         Ok(())
     }
 
+    pub(crate) fn do_bsp_confirm_storing(
+        who: StorageProviderId<T>,
+        location: FileLocation<T>,
+        root: FileKey<T>,
+        proof: Proof<T>,
+    ) -> DispatchResult {
+        let bsp = match <T::Providers as storage_hub_traits::ReadProvidersInterface>::get_provider(
+            who.clone(),
+        ) {
+            Some(bsp) => bsp,
+            None => return Err(Error::<T>::NotABsp.into()),
+        };
```

Comment:
Why using `expect_or_err` here? `location` is a parameter passed by the user, it is perfectly possible that it is wrong.

Reply:
true!